src/core/lib/iomgr/error.h

```diff
@@ -272,4 +272,11 @@ inline bool grpc_log_if_error(const char* what, grpc_error* error,
 #define GRPC_LOG_IF_ERROR(what, error) \
   (grpc_log_if_error((what), (error), __FILE__, __LINE__))
 
+typedef struct {
```

Comment:
I'm really not enthusiastic about exposing this as part of the error API.  This should really be restricted to being an internal implementation detail.  I would strongly prefer to keep this in the .cc file and have callers use `grpc_error_get_str()`.

Is that going to undo the whole benefit of this change?

Reply:
Per your other comment, I think we can avoid this.